chroot/pty_ptmx.go

```diff
@@ -0,0 +1,53 @@
+//go:build linux
+// +build linux
+
+package chroot
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// Open a PTY using the /dev/ptmx device. The main advantage of using
+// this instead of posix_openpt is that it avoids cgo.
+func getPtyDescriptors() (int, int, error) {
+	// Create a pseudo-terminal -- open a copy of the master side.
+	controlFd, err := unix.Open("/dev/ptmx", os.O_RDWR, 0600)
+	if err != nil {
+		logrus.Errorf("error opening PTY master using /dev/ptmx: %v", err)
```

Comment:
Do not logrus. Errorf here, Just wrap the err below.

return -1, -1, fmt.Errorf("error opening PTY master using /dev/ptmx: %w", err)